src/NHSD.BuyingCatalogue.Identity.Api/Controllers/AccountController.cs

```diff
@@ -92,20 +89,37 @@ public async Task<IActionResult> Login(LoginViewModel viewModel)
             if (Url.IsLocalUrl(returnUrl))
                 return Redirect(returnUrl);
 
-            if (string.IsNullOrEmpty(returnUrl))
+            if (string.IsNullOrWhiteSpace(returnUrl))
                 return Redirect("~/");
 
-            // User might have clicked on a malicious link - should be logged
-            throw new Exception("Invalid return URL");
+            // Return URL could be malicious
+            throw new InvalidReturnUrlException();
         }
 
-        [HttpGet]
         public async Task<IActionResult> Error(string errorId)
```

Comment:
At this stage of development, `errorId` will always be `null` so we do we actually require it?

This also affects the design of the error view model – please see the comments below about that.

Reply:
If we don't need it, get rid of it.